lib/geocoder/lookups/olleh.rb

```diff
@@ -135,20 +138,23 @@ def results(query)
       end
 
       case Olleh.check_query_type(query)
-      when "geocoding"
+      when 'addr_local_search'
+        return [] if doc["RESULTDATA"]["place"]["TotalCount"].to_i == 0
+        return doc["RESULTDATA"]["place"]["Data"]
+      when 'geocoding'
         return [] if doc['RESDATA']['COUNT'] == 0
-        return doc['RESDATA']["ADDRS"]
-      when "reverse_geocoding"
+        return doc['RESDATA']['ADDRS']
+      when 'reverse_geocoding'
         return [] if doc['RESDATA']['COUNT'] == 0
-        return doc['RESDATA']["ADDRS"] || []
-      when "route_search"
-        return [] if doc["RESDATA"]["SROUTE"]["isRoute"] == "false"
-        return doc["RESDATA"] || []
-      when "convert_coord"
+        return doc['RESDATA']['ADDRS'] || []
+      when 'route_search'
+        return [] if doc['RESDATA']['SROUTE']['isRoute'] == 'false'
         return doc['RESDATA'] || []
-      when "addr_step_search"
+      when 'convert_coord'
+        return doc['RESDATA'] || []
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.